activesupport/lib/active_support/callbacks.rb

```diff
@@ -286,9 +286,8 @@ def self.simple(callback_sequence, user_callback)
       class Callback #:nodoc:#
         def self.build(chain, filter, kind, options)
           if filter.is_a?(String)
-            ActiveSupport::Deprecation.warn(<<-MSG.squish)
-              Passing string to define callback is deprecated and will be removed
-              in Rails 5.1 without replacement.
+            raise ArgumentError, <<-MSG.squish
+              Passing string to define callback is not supported. Use proc instead.
             MSG
           end
```

Comment:
@rafaelfranca @matthewd is the original deprecation warning obtuse enough that we could construe it to mean `…also don't use strings in :if and :unless`?